number_theory/modular.h

```diff
@@ -148,85 +148,85 @@ struct is_modular {
 }  // namespace modular_internal
 
 template <typename T>
-concept ModularType = modular_internal::is_modular<T>::value;
+concept ModularType = modular_internal::ModularTypeImpl<T>::value;
 
 // Overloads an arithmetic operator with a class method.
-#define OVERLOAD_ARITHMETIC_OPERATOR(CONCEPT, OP, METHOD) \
-  template <CONCEPT M>                                    \
-  M operator OP(const M &lhs, const M &rhs) {             \
-    return lhs.METHOD(rhs);                               \
-  }                                                       \
-  template <CONCEPT M, std::convertible_to<M> T>          \
-  M operator OP(const M &lhs, const T &rhs) {             \
-    return lhs.METHOD(static_cast<M>(rhs));               \
-  }                                                       \
-  template <CONCEPT M, std::convertible_to<M> T>          \
-  M operator OP(const T &lhs, const M &rhs) {             \
-    return static_cast<M>(lhs).METHOD(rhs);               \
+#define OVERLOAD_MODULAR_ARITHMETIC_OPERATOR(OP, METHOD) \
+  template <ModularType M>                               \
```

Comment:
I wonder if using `M` as typename good or not. In chromium's code base, they mostly use `T` for the first typename and `U` for the second one. However, they are always `typename T` whereas ours is `ModularType M`. In this case, it actually makes sense. I just wanted to point this out before we merge it.

Reply:
I think the reason is that chromium does not use c++20.